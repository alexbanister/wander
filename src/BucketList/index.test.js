import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import BucketListContainer, { BucketList } from './index';
import React from 'react';


describe('BucketListContainer', () => {
  it('should always match the snapshot', () => {
    const mockStore = configureStore();
    const initialState = {};
    const store = mockStore(initialState);
    const wrapper = shallow(<BucketListContainer
      store = {store}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('BucketList', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<BucketList
      destinations={[]}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should change state', () => {
    const expected = {
      destination: 'DE',
      airports: []
    };

    const wrapper = shallow(<BucketList
      destinations={[]}
    />);

    expect(wrapper).toMatchSnapshot();

    const destination = wrapper.find('[type="text"]');

    destination.simulate('change', {target: {value: 'DE'}});

    expect(wrapper.state()).toEqual(expected);
  });
});
