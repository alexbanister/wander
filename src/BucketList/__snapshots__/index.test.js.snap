// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BucketList should change state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BucketList
    destinations={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
>
          <input
                    onChange={[Function]}
                    placeholder="Add Destination"
                    type="text"
                    value=""
          />
</form>,
        <div
          className="airportListContainer"
>
          <div
                    className="airportList"
          >
                    
          </div>
</div>,
        <div
          className="destinationList"
>
          
</div>,
        <button
          onClick={[Function]}
>
          Search All Destinations
</button>,
      ],
      "className": "bucketlist",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <input
            onChange={[Function]}
            placeholder="Add Destination"
            type="text"
            value=""
/>,
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Add Destination",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "form",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="airportList"
>
            
</div>,
          "className": "airportListContainer",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "airportList",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "destinationList",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Search All Destinations",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Search All Destinations",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
>
            <input
                        onChange={[Function]}
                        placeholder="Add Destination"
                        type="text"
                        value=""
            />
</form>,
          <div
            className="airportListContainer"
>
            <div
                        className="airportList"
            >
                        
            </div>
</div>,
          <div
            className="destinationList"
>
            
</div>,
          <button
            onClick={[Function]}
>
            Search All Destinations
</button>,
        ],
        "className": "bucketlist",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <input
              onChange={[Function]}
              placeholder="Add Destination"
              type="text"
              value=""
/>,
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Add Destination",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "form",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="airportList"
>
              
</div>,
            "className": "airportListContainer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "airportList",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "destinationList",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Search All Destinations",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Search All Destinations",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`BucketList should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BucketList
    destinations={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
>
          <input
                    onChange={[Function]}
                    placeholder="Add Destination"
                    type="text"
                    value=""
          />
</form>,
        <div
          className="airportListContainer"
>
          <div
                    className="airportList"
          >
                    
          </div>
</div>,
        <div
          className="destinationList"
>
          
</div>,
        <button
          onClick={[Function]}
>
          Search All Destinations
</button>,
      ],
      "className": "bucketlist",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <input
            onChange={[Function]}
            placeholder="Add Destination"
            type="text"
            value=""
/>,
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Add Destination",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "form",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="airportList"
>
            
</div>,
          "className": "airportListContainer",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "airportList",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "destinationList",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Search All Destinations",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Search All Destinations",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
>
            <input
                        onChange={[Function]}
                        placeholder="Add Destination"
                        type="text"
                        value=""
            />
</form>,
          <div
            className="airportListContainer"
>
            <div
                        className="airportList"
            >
                        
            </div>
</div>,
          <div
            className="destinationList"
>
            
</div>,
          <button
            onClick={[Function]}
>
            Search All Destinations
</button>,
        ],
        "className": "bucketlist",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <input
              onChange={[Function]}
              placeholder="Add Destination"
              type="text"
              value=""
/>,
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Add Destination",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "form",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="airportList"
>
              
</div>,
            "className": "airportListContainer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "airportList",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "destinationList",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Search All Destinations",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Search All Destinations",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`BucketListContainer should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(Connect(BucketList))
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
